### PR TITLE
MOSIP-13282

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: registration-processor/pre-processor/registration-processor-cmd-validator-stage/src/main/java/io/mosip/registration/processor/stages/app/CMDValidationProcessor.java [hunk 1]
```diff
@@ -146,20 +160,37 @@ public MessageDTO process(MessageDTO object, String stageName) {
...
+			boolean isSubProcess=false;
+			if(subProcess!=null || !subProcess.isBlank()) {
+			 isSubProcess=Arrays.asList(subProcess.split(",")).contains(registrationStatusDto.getRegistrationType());
 			}
-
-			if (validateMachine) {
-				machineValidator.validate(regOsi.getMachineId(), primaryLanguagecode, regOsi.getPacketCreationDate(),
-						registrationStatusDto.getRegistrationId());
+			if(isSubProcess) {
+				if (validateCenter && validateCenterForSubProcess) {
```
Comment (vishwa-vyom) [inline registration-processor/pre-processor/registration-processor-cmd-validator-stage/src/main/java/io/mosip/registration/processor/stages/app/CMDValidationProcessor.java]: If there are 2 sub process 1. biometric correction which needs device validation to be done 2. document correction which does not require the device validation, how can we switch off for 1 and not for other ?

File: registration-processor/pre-processor/registration-processor-introducer-validator-stage/src/main/java/io/mosip/registration/processor/stages/introducervalidator/IntroducerValidator.java [hunk 2]
```diff
@@ -69,6 +71,15 @@
...
 	private Utilities utility;
+	
+	@Value("registration.processor.sub-process")
+    private String subProcess;
+
+	@Value("${mosip.registration.processor.validate-IntroducerID-for-sub-process:true}")
+	private boolean validateIntroducerIDForSubProcess;
+
+	@Value("${mosip.registration.processor.validate-IntroducerBIOMETRIC-for-sub-process:true}")
+	private boolean validateIntroducerBIOMETRICForSubProcess;
```
Comment (vishwa-vyom) [inline registration-processor/pre-processor/registration-processor-introducer-validator-stage/src/main/java/io/mosip/registration/processor/stages/introducervalidator/IntroducerValidator.java]: Why do we need this change for introducer validation, in a sub process or main process if intorducer validation is not required, then they will not even call this stage.

**Review by vishwa-vyom — CHANGES_REQUESTED** [review]

File: registration-processor/pre-processor/registration-processor-cmd-validator-stage/src/main/java/io/mosip/registration/processor/stages/app/CMDValidationProcessor.java [hunk 3]
```diff
@@ -1,259 +1,260 @@
...
+	private OSIUtils osiUtils;
+
+	@Value("${mosip.registration.gps_device_enable_flag}")
+	private String gpsEnable;
+
+	@Value("${mosip.primary-language}")
+	private String primaryLanguagecode;
+
+	@Value("#{'${mosip.regproc.cmd.center.validation.processes:CORRECTION,NEW,UPDATE,LOST,BIOMETRIC_CORRECTION,DOCUMENT_CORRECTION,DEMOGRAPHIC_CORRECTION}'.split(',')}")
+	private List<String> centerSubProcesses;
```
Comment (vishwa-vyom) [inline registration-processor/pre-processor/registration-processor-cmd-validator-stage/src/main/java/io/mosip/registration/processor/stages/app/CMDValidationProcessor.java]: Please change the property to mosip.regproc.cmd-validator.center-validation.processes:NEW,UPDATE,LOST,BIOMETRIC_CORRECTION

We can only have the processes that we are going to have as samples

Also please change the variable name to centerValidationProcessList (This is not only sub process)

Follow this config and variable naming conventation for other 2 properties also

File: registration-processor/pre-processor/registration-processor-cmd-validator-stage/src/main/java/io/mosip/registration/processor/stages/app/CMDValidationProcessor.java [hunk 4]
```diff
@@ -1,259 +1,260 @@
...
+	
+	@Value("#{'${mosip.regproc.cmd.machine.validation.processes:CORRECTION,NEW,UPDATE,LOST,BIOMETRIC_CORRECTION,DOCUMENT_CORRECTION,DEMOGRAPHIC_CORRECTION}'.split(',')}")
```
Comment (vishwa-vyom) [inline registration-processor/pre-processor/registration-processor-cmd-validator-stage/src/main/java/io/mosip/registration/processor/stages/app/CMDValidationProcessor.java]: did you test with a empty string for the property ? did the list was empty or there was 1 item in the list with a empty string ?

**Review by vishwa-vyom — CHANGES_REQUESTED** [review]

File: registration-processor/pre-processor/registration-processor-cmd-validator-stage/src/main/java/io/mosip/registration/processor/stages/app/CMDValidationProcessor.java (hunk 3 above)
Comment (vishwa-vyom) [reply registration-processor/pre-processor/registration-processor-cmd-validator-stage/src/main/java/io/mosip/registration/processor/stages/app/CMDValidationProcessor.java]: Property name is not maching with what I have commented above

Comment (ghost) [reply registration-processor/pre-processor/registration-processor-cmd-validator-stage/src/main/java/io/mosip/registration/processor/stages/app/CMDValidationProcessor.java]: I have changed the properties as requested

File: registration-processor/pre-processor/registration-processor-cmd-validator-stage/src/main/java/io/mosip/registration/processor/stages/app/CMDValidationProcessor.java (hunk 4 above)
Comment (ghost) [reply registration-processor/pre-processor/registration-processor-cmd-validator-stage/src/main/java/io/mosip/registration/processor/stages/app/CMDValidationProcessor.java]: Yes I have tested with empty string and have added test case for it

**Review by vishwa-vyom — APPROVED** [review]

**Pull request merged**